backend/services/unified_samurai_agent.py

```
@@ -305,18 +304,45 @@ async def _analyze_user_intent(
+                    f"Description: {getattr(context.task_context, 'description', '')}\n"
+                    f"Status: {getattr(context.task_context, 'status', 'pending')}\n"
+                    f"Priority: {getattr(context.task_context, 'priority', 'medium')}\n\n"
+                    "IMPORTANT NOTE: The user explicitly selected this active task. ALL generated tasks must strictly advance THIS task.\n"
+                    "- Create ONLY subtasks for this active task. Do NOT create a new root task.\n"
+                    "- Use conversation history only insofar as it informs or unblocks this active task.\n"
+                    "- Ignore unrelated conversation topics unless the user explicitly asked to expand scope.\n\n"
+                )
+            # Guidance when there is no active task: infer target task from latest conversation
+            no_active_task_inference = ""
+            if not context.task_context:
+                no_active_task_inference = (
```

Ruff (0.12.2)

Local variable `no_active_task_inference` is assigned to but never used

Remove assignment to unused variable `no_active_task_inference`

(F841)